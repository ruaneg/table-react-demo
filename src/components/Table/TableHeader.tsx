import { StyledTableHeader, StyledTableRow } from './Table.styles';

export const TableHeader = ({ headers }: { headers: string[] }) => {
  return (
    <thead>
      <StyledTableRow>
        {headers.map((header, index) => (
          <StyledTableHeader key={index}>{header}</StyledTableHeader>
        ))}
      </StyledTableRow>
    </thead>
  );
};
