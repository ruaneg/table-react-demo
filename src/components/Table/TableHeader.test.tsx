import { render } from '@testing-library/react';
import 'jest-styled-components';
import { describe, test, expect } from 'vitest';
import { TableHeader } from './TableHeader';

describe('TableHeader', () => {
  test('renders correctly with given headers', () => {
    const headers = ['Name', 'Age', 'Address'];
    const { container } = render(<TableHeader headers={headers} />);
    expect(container.firstChild).toMatchSnapshot();
  });

  test('renders the correct number of headers', () => {
    const headers = ['Name', 'Age', 'Address'];
    const { getAllByRole } = render(<TableHeader headers={headers} />);
    const headerElements = getAllByRole('columnheader');
    expect(headerElements).toHaveLength(headers.length);
  });

  test('renders headers with correct text', () => {
    const headers = ['Name', 'Age', 'Address'];
    const { getAllByRole } = render(<TableHeader headers={headers} />);
    const headerElements = getAllByRole('columnheader');
    headerElements.forEach((headerElement, index) => {
      expect(headerElement).toHaveTextContent(headers[index]);
    });
  });

  test.skip('applies correct styles to headers and rows', () => {
    const headers = ['Name', 'Age', 'Address'];
    const { container } = render(<TableHeader headers={headers} />);
    const row = container.querySelector('tr');
    const header = container.querySelector('th');

    expect(row).toHaveStyleRule('display', 'table-row');
    expect(header).toHaveStyleRule('display', 'table-cell');
  });
});
