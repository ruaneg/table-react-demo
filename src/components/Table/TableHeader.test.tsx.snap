// Vitest Snapshot v1

exports[`TableHeader > renders correctly with given headers 1`] = `
.c1 {
  padding: 8px;
  border: 1px solid #f1efef;
  border-left: none;
  border-right: none;
  font-size: 1rem;
  font-weight: 500;
  text-align: left;
}

.c0 {
  background: white;
  border: 1px solid #f1efef;
}

.c0:hover {
  background: #f1f1f1;
}

<thead>
  <tr
    class="c0"
  >
    <th
      class="c1"
    >
      Name
    </th>
    <th
      class="c1"
    >
      Age
    </th>
    <th
      class="c1"
    >
      Address
    </th>
  </tr>
</thead>
`;
